src/Model/Repository/IssueAssociationRepository.php

```diff
@@ -0,0 +1,197 @@
+
+        Entity\IssueAssociation::create()
+            ->setIssueId($issue_id)
+            ->setAssociatedId($associated_issue_id)
+            ->save();
+
+        History::add(
+            $issue_id, $usr_id, 'issue_associated', 'Issue associated to Issue #{associated_id} by {user}', [
+                'associated_id' => $associated_issue_id,
+                'user' => User::getFullName($usr_id)
+            ]
+        );
+    }
+
+    /**
+     * Method used to remove an issue association from an issue.
+     *
+     * @param int $usr_id User Id performing the operation
+     * @param int $issue_id The issue ID
+     * @param int $associated_issue_id The associated issue ID to remove.
+     */
+    public function removeAssociation($usr_id, $issue_id, $associated_issue_id)
+    {
+        // see if there already is association
+        $assoc = $this->getAssociatedIssues($issue_id);
+        if (!in_array($associated_issue_id, $assoc)) {
+            throw new InvalidArgumentException("Issue $issue_id not associated to $associated_issue_id");
+        }
+
+        Entity\IssueAssociation::create()->removeAssociation($issue_id, $associated_issue_id);
+
+        $full_name = User::getFullName($usr_id);
+        $pairs = [
+            [$issue_id, $associated_issue_id],
+            [$associated_issue_id, $issue_id],
+        ];
+
+        foreach ($pairs as $pair) {
+            list($issue_id, $associated_issue_id) = $pair;
+            $params = [
+                'issue_id' => $associated_issue_id,
+                'user' => $full_name
+            ];
+            History::add(
+                $issue_id, $usr_id, 'issue_unassociated',
+                'Issue association to Issue #{issue_id} removed by {user}', $params
+            );
+        }
+    }
+
+    /**
+     * Update the issue associations
+     *
+     * @param int $usr_id User Id performing the operation
+     * @param int $issue_id issue_id to update associations
+     * @param int[] $issues issue_id's to associate with
+     * @return string[] errors from operation
+     */
+    public function updateAssociations($usr_id, $issue_id, $issues)
+    {
+        list($issues, $errors) = $this->filterExistingIssues($issues, $issue_id);
+        $existing_associations = $this->getAssociatedIssues($issue_id);
+
+        $add = array_diff($issues, $existing_associations);
+        $remove = array_diff($existing_associations, $issues);
+        if (!$add && !$remove) {
+            return [];
+        }
+
+        foreach ($add as $associated_id) {
+            $this->addIssueAssociation($usr_id, $issue_id, $associated_id);
+        }
+        foreach ($remove as $associated_id) {
+            $this->removeAssociation($usr_id, $issue_id, $associated_id);
+        }
+
+        return $errors;
+    }
+
+    /**
+     * Filter issues for invalid input and issues that do not exist.
+     *
+     * XXX: check_project was false in old code even if it said it does check for project
```

Comment:
If it did nothing in the old code I'm happy to have it removed. Technically you are leaking information from one project to the other, but only that an issue exists and if it is closed or not. Otherwise we need to add some sort of preference to allow/prevent cross-project issue association.